c++/src/kj/test-helpers.c++

```diff
@@ -28,17 +32,35 @@
 #include <process.h>
 #endif
 
+#include <functional>
+
 namespace kj {
 namespace _ {  // private
 
 bool hasSubstring(StringPtr haystack, StringPtr needle) {
-  // TODO(perf): This is not the best algorithm for substring matching.
   if (needle.size() <= haystack.size()) {
+    // Boyer Moore Horspool wins https://quick-bench.com/q/RiKdKduhdLb6x_DfS1fHaksqwdQ
```

Comment:
Well, you've done the work now, but I feel like this was a lot of effort spent optimizing a test helper that only ever runs on small strings anyway...

Reply:
Yeah... had some time to kill waiting for CI :).